src/main/kotlin/com/deviceinsight/helm/AbstractHelmMojo.kt

```diff
@@ -16,46 +16,161 @@
 
+import com.deviceinsight.helm.model.Registry
+import com.deviceinsight.helm.model.Repo
+import com.deviceinsight.helm.model.validate
+import com.deviceinsight.helm.util.HelmDownloader
+import com.deviceinsight.helm.util.PlatformDetector
+import com.deviceinsight.helm.util.ServerAuthentication
+import org.apache.maven.artifact.repository.ArtifactRepository
+import org.apache.maven.artifact.resolver.ArtifactResolutionRequest
 import org.apache.maven.plugin.AbstractMojo
+import org.apache.maven.plugins.annotations.Component
 import org.apache.maven.plugins.annotations.Parameter
 import org.apache.maven.project.MavenProject
+import org.apache.maven.repository.RepositorySystem
+import org.apache.maven.settings.Settings
+import org.sonatype.plexus.components.sec.dispatcher.SecDispatcher
 import java.io.File
+import java.net.URI
 import kotlin.concurrent.thread
 
-abstract class AbstractHelmMojo : AbstractMojo() {
+abstract class AbstractHelmMojo : AbstractMojo(), ServerAuthentication {
+	@Parameter(property = "skip", defaultValue = "false")
+	private var skip: Boolean = false
 
-	@Parameter(property = "chartVersion", required = false, defaultValue = "\${project.model.version}")
+	@Parameter(property = "chartVersion", required = true, defaultValue = "\${project.model.version}")
 	protected lateinit var chartVersion: String
 
-	@Parameter(defaultValue = "\${project}", readonly = true, required = true)
+	@Parameter(property = "chartName", required = true, defaultValue = "\${project.artifactId}")
+	protected lateinit var chartName: String
+
+	@Parameter(property = "chartFolder", required = true, defaultValue = "src/main/helm/\${chartName}")
+	protected lateinit var chartFolder: String
+
+	@Parameter(property = "helmGroupId", required = true, defaultValue = "com.deviceinsight.helm")
+	private lateinit var helmGroupId: String
+
+	@Parameter(property = "helmArtifactId", required = true, defaultValue = "helm")
+	private lateinit var helmArtifactId: String
+
+	@Parameter(property = "helmVersion", required = true, defaultValue = "3.11.3")
+	private lateinit var helmVersion: String
+
+	@Parameter(property = "helmDownloadUrl", required = true, defaultValue = "https://get.helm.sh/")
+	private lateinit var helmDownloadUrl: URI
+
+	@Parameter(property = "repos")
+	protected var repos: List<Repo> = emptyList()
+
+	@Parameter(property = "registries")
+	protected var registries: List<Registry> = emptyList()
+
+	@Parameter(readonly = true, required = true, defaultValue = "\${localRepository}")
+	private lateinit var localRepository: ArtifactRepository
+
+	@Parameter(readonly = true, required = true, defaultValue = "\${project.remoteArtifactRepositories}")
+	private lateinit var remoteRepositories: List<ArtifactRepository>
+
+	@Component(role = SecDispatcher::class, hint = "default")
+	override lateinit var securityDispatcher: SecDispatcher
+
+	@Parameter(defaultValue = "\${settings}", readonly = true)
+	override lateinit var settings: Settings
+
+	@Component
```

Comment:
The maven-plugin-plugin emits a warning and says this is deprecated and should be replaced with
```suggestion
	@Parameter(defaultValue = "\${project}", readonly = true)
```